sdk/file_sdk/UnitTests/xmp_ut/xmp_file_tests.cpp

```diff
@@ -14,28 +21,94 @@ class Xmp_Tests : public QObject
   Xmp_Tests();
 
 private Q_SLOTS:
-  void GetTags_FileWithTag_ReturnCorrectTag();
+  void GetTags_FileWithAutomaticTag_ReturnCorrectTag();
+  void GetTags_FileWithManualTag_ReturnCorrectTag();
 };
 
-Xmp_Tests::Xmp_Tests()
-{
+Xmp_Tests::Xmp_Tests() {
 }
 
-void Xmp_Tests::GetTags_FileWithTag_ReturnCorrectTag()
-{
-  /*mip::Tag general( "f42aa342-8706-4288-bd11-ebb85995028c", "General", "", "vakuras@microsoft.com", true, "", "", mip::Method::MANUAL,"72f988bf-86f1-41af-91ab-2d7cd011db47" );
+bool VerifyTags(Tag tag1, Tag tag2) {
+  if (tag1.GetExtendedProperties().size() != tag2.GetExtendedProperties().size())
+    return false;
+
+  for (size_t i = 0; i < tag1.GetExtendedProperties().size(); i++) {
+    if (tag1.GetExtendedProperties()[i].key != tag2.GetExtendedProperties()[i].key ||
+        tag1.GetExtendedProperties()[i].value != tag2.GetExtendedProperties()[i].value ||
+        tag1.GetExtendedProperties()[i].vendor != tag2.GetExtendedProperties()[i].vendor) {
+      return false;
+    }
+  }
+
+  return
+      tag1.GetLabelId() == tag2.GetLabelId() &&
+      tag1.GetLabelName() == tag2.GetLabelName() &&
+      tag1.GetLabelParentId() == tag2.GetLabelParentId() &&
+      tag1.GetEnabled() == tag2.GetEnabled() &&
+      tag1.GetOwner() == tag2.GetOwner() &&
+      tag1.GetSiteId() == tag2.GetSiteId();

```

Comment:
check Method

Reply:
i check the method in the extended properties because we moved it.
I will check the SetTime for one file

---
In reply to: [136715671](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r136715671) [](ancestors = 136715671)